variables.tf

```
@@ -42,6 +42,106 @@ variable "helm_additional_values" {
   default     = []
 }
 
+variable "node_affinity" {
+  description = "Node affinity settings for Kyverno pods. Use weight as map key; operator is `In` and policy is `preferredDuringSchedulingIgnoredDuringExecution`."
+  type = map(object({
+    key    = string
+    values = list(string)
+  }))
+  default = null
+}
+
+variable "admissioncontroller_node_affinity" {
+  description = "Node affinity settings for admissioncontroller pods. Use weight as map key; operator is `In` and policy is `preferredDuringSchedulingIgnoredDuringExecution`."
+  type = map(object({
+    key    = string
+    values = list(string)
+  }))
+  default = null
+}
+
+variable "backgroundcontroller_node_affinity" {
+  description = "Node affinity settings for backgroundcontroller pods. Use weight as map key; operator is `In` and policy is `preferredDuringSchedulingIgnoredDuringExecution`."
+  type = map(object({
+    key    = string
+    values = list(string)
+  }))
+  default = null
+}
+
+variable "cleanupcontroller_node_affinity" {
+  description = "Node affinity settings for cleanupcontroller pods. Use weight as map key; operator is `In` and policy is `preferredDuringSchedulingIgnoredDuringExecution`."
+  type = map(object({
+    key    = string
+    values = list(string)
+  }))
+  default = null
+}
+
+variable "reportscontroller_node_affinity" {
+  description = "Node affinity settings for reportscontroller pods. Use weight as map key; operator is `In` and policy is `preferredDuringSchedulingIgnoredDuringExecution`."
+  type = map(object({
+    key    = string
+    values = list(string)
+  }))
+  default = null
+}
+
+variable "tolerations" {
+  description = "Tolerations for Kyverno pods."
```

```suggestion
  description = "Tolerations for Kyverno deployments."
```

We could maybe add something in the description that specifies that this configuration is used in the `global` helm values and something in the specific tolerations that specifies that they are used for the specific component.